src/Core/Components/Pagination/FluentPaginator.razor.cs

```diff
@@ -0,0 +1,95 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components.Infrastructure;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// A component that provides a user interface for <see cref="PaginationState"/>.
+/// </summary>
+public partial class FluentPaginator : FluentComponentBase, IDisposable
+{
+    private readonly EventCallbackSubscriber<PaginationState> _totalItemCountChanged;
+    private readonly EventCallbackSubscriber<PaginationState> _currentPageItemsChanged;
+
+    /// <summary>
+    /// Gets or sets the callback that is invoked when the current page index changes.
+    /// </summary>
+    /// <remarks>The callback receives the new page index as an <see cref="int"/> parameter. Use this property
+    /// to handle page index changes in a parent component or service.</remarks>
+    [Parameter]
+    public EventCallback<int> CurrentPageIndexChanged { get; set; }
+
+    /// <summary>
+    /// Disables the pagination buttons
+    /// </summary>
+    [Parameter]
+    public bool Disabled { get; set; }
+
+    /// <summary>
+    /// Gets or sets the associated <see cref="PaginationState"/>. This parameter is required.
+    /// </summary>
+    [Parameter, EditorRequired]
+    public PaginationState State { get; set; } = default!;
+
+    /// <summary>
+    /// Optionally supplies a template for rendering the page count summary.
+    /// The following values can be included:
+    /// {your State parameter name}.TotalItemCount (for the total number of items)
+    /// </summary>
+    [Parameter]
+    public RenderFragment? SummaryTemplate { get; set; }
+
+    /// <summary>
+    /// Optionally supplies a template for rendering the pagination summary.
+    /// The following values can be included:
+    /// {your State parameter name}.CurrentPageIndex (zero-based, so +1 for the current page number)
+    /// {your State parameter name}.LastPageIndex (zero-based, so +1 for the total number of pages)
+    /// </summary>
+    [Parameter]
+    public RenderFragment? PaginationTextTemplate { get; set; }
+
+    /// <summary>
+    /// Constructs an instance of <see cref="FluentPaginator" />.
+    /// </summary>
+    public FluentPaginator()
```

Comment:
Can you move the ctor before the properties?
We can respect the order:
- Private variables
- Constructors
- properties
- Methods
- private methods